### PR TITLE
Add map compass to reset rotation

### LINKED ISSUE #28: Add compass button/icon to reorient
Check out google maps and apple maps for implementations.

### PR DESCRIPTION
**Pull request by bforcum**
Add compass button under the settings button to reorient the map to north-up

Lock device orientation to north-up

Closes #28

### CODE REVIEW

File: lib/page/home_page/map_compass.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        child: IconButton(
+          alignment: Alignment.center,
+          padding: EdgeInsets.zero,
+          icon: Stack(
+            alignment: Alignment.center,
+            children: [
+              Transform.rotate(
+                angle: degToRadian(camera.rotation + widget.rotationOffset),
+                child: widget.icon ?? PlatformWidget(
+                  cupertino: (_, __) => SizedBox(height: 50, child: Image(image: View.of(context).platformDispatcher.platformBrightness == Brightness.light
```
Comment (elliotnash) [inline lib/page/home_page/map_compass.dart]: This doesn't respect the lightmode/darkmode setting, instead use `Theme.of(context).brightness`

File: lib/main.dart [hunk 2]
```diff
@@ -64,6 +65,11 @@ class _AppState extends ConsumerState<App> with WidgetsBindingObserver {
 
   @override
   Widget build(BuildContext context) {
+    SystemChrome.setPreferredOrientations(
+      [
+        DeviceOrientation.portraitUp
```
Comment (elliotnash) [inline lib/main.dart:69]: Does this lock the device to portrait mode?

File: lib/main.dart [hunk 3]
```diff
@@ -4,6 +4,7 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter_platform_widgets/flutter_platform_widgets.dart';
+import 'package:flutter_uploader/flutter_uploader.dart';
```
Comment (elliotnash) [inline lib/main.dart]: Remove unused import

**Review by elliotnash — CHANGES_REQUESTED** [review]
Looking great! A few small things

File: lib/main.dart (hunk 2 above)
Comment (bforcum) [reply lib/main.dart:69]: Yes. I could make that only apply to mobile platforms if you want.

Comment (elliotnash) [reply lib/main.dart:69]: it seems great how it is

**Review by elliotnash — APPROVED** [review]

**Pull request merged**